app/src/main/java/io/apicurio/registry/storage/impl/sql/SQLServerSqlStatements.java

```diff
@@ -31,83 +31,121 @@ public SQLServerSqlStatements() {
     }
 
     /**
-     * @see SqlStatements#dbType()
+     * @see io.apicurio.registry.storage.impl.sql.SqlStatements#dbType()
      */
     @Override
     public String dbType() {
         return "mssql";
     }
 
     /**
-     * @see SqlStatements#isPrimaryKeyViolation(Exception)
+     * @see io.apicurio.registry.storage.impl.sql.SqlStatements#isPrimaryKeyViolation(java.lang.Exception)
      */
     @Override
     public boolean isPrimaryKeyViolation(Exception error) {
-        return error.getMessage().contains("violates unique constraint");
+        return error.getMessage().contains("Violation of PRIMARY KEY constraint");
     }
 
     /**
-     * @see SqlStatements#isForeignKeyViolation(Exception)
+     * @see io.apicurio.registry.storage.impl.sql.SqlStatements#isForeignKeyViolation(java.lang.Exception)
      */
     @Override
     public boolean isForeignKeyViolation(Exception error) {
-        return error.getMessage().contains("violates foreign key constraint");
+        return error.getMessage().contains("conflicted with the FOREIGN KEY constraint");
     }
 
     /**
-     * @see SqlStatements.core.storage.jdbc.ISqlStatements#isDatabaseInitialized()
+     * @see io.apicurio.registry.storage.impl.sql.SqlStatements.core.storage.jdbc.ISqlStatements#isDatabaseInitialized()
      */
     @Override
     public String isDatabaseInitialized() {
         return "SELECT count(*) AS count FROM information_schema.tables WHERE table_name = 'artifacts'";
     }
 
     /**
-     * @see SqlStatements#upsertContent()
+     * @see io.apicurio.registry.storage.impl.sql.SqlStatements#upsertContent()
      */
     @Override
     public String upsertContent() {
-        return "INSERT INTO content (tenantId, contentId, canonicalHash, contentHash, content, artifactreferences) VALUES (?, ?, ?, ?, ?, ?) ON CONFLICT (tenantId, contentHash) DO NOTHING";
+        return String.join(" ",
+                "MERGE INTO content AS target",
+                "USING (VALUES (?, ?, ?, ?, ?, ?)) AS source (tenantId, contentId, canonicalHash, contentHash, content, artifactreferences)",
+                "ON (target.tenantId = source.tenantId AND target.contentHash = source.contentHash)",
+                "WHEN NOT MATCHED THEN",
+                    "INSERT (tenantId, contentId, canonicalHash, contentHash, content, artifactreferences)",
+                    "VALUES (source.tenantId, source.contentId, source.canonicalHash, source.contentHash, source.content, source.artifactreferences);");
```

Comment:
I like this use of String.join - I hadn't thought to use it this way.  But can we **please** get multi-line string literals in Java asap? :)